### ISSUE TITLE
How to set and use environment variable in docker-compose.yml?

### ISSUE THREAD

**Issue by tvaughan** [docker/compose#3208]
I'm trying to specify the value of a variable only once. For example:

```
$ cat Dockerfile 
FROM ubuntu:1510
MAINTAINER "Tom Vaughan <tvaughan@tocino.cl>"

ARG SOME_PORT
ENV SOME_PORT $SOME_PORT

EXPOSE $SOME_PORT

$ cat docker-compose.yml 
version: "2"
services:
  example:
    environment:
      - SOME_PORT="3000"
    build:
      context: .
      args:
        - SOME_PORT=$SOME_PORT
    ports:
      - $SOME_PORT:$SOME_PORT

$ docker-compose config
WARNING: The SOME_PORT variable is not set. Defaulting to a blank string.
ERROR: Validation failed in file './docker-compose.yml', reason(s):
services.example.ports is invalid: Invalid port ":", should be [[remote_ip:]remote_port[-remote_port]:]port[/protocol]
```

```
MacOS X El Capitan
$ docker-compose --version
docker-compose version 1.6.2, build 4d72027
```

What am I doing wrong? Thanks!

**Comment by dnephin** [comment]
`environment` only applies to the container environment not the host environment.

We just released 1.7.0-rc1 which supports a `.env` file which lets you see default values for variables on the host.

**Comment by tvaughan** [comment]
Cool. This seems to work. Thanks!
